lib/api-client.js

```diff
@@ -116,6 +116,10 @@ exports.newNutshell = function (opt, callback) {
             } else {
               Array.prototype.push.apply(args.state.results, data.result);
             }
+
+            if (!data.result) {
+              return cb("no result from nutshell");
```

Comment:
I think it would be better to return an Error object rather than just a string.  Something like:

```js
return cb(new Error('No result from nutshell'));
```